definitions/npm/react-redux_v7.x.x/flow_v0.142.x-/react-redux_v7.x.x.js

```diff
@@ -200,7 +200,15 @@ declare module "react-redux" {
   // Typings for Hooks
   // ------------------------------------------------------------
 
-  declare export function useDispatch<D>(): D;
+  declare export function useDispatch<D>(): (
+    & (<T: { [key: string]: any }>(T) => T)
+    // Supports thunks at their various lengths and use cases depending if user has typed them as tuple vs array
+    & (<T>((...args: [any]) => T) => T)
+    & (<T>((...args: [any, any]) => T) => T)
+    & (<T>((...args: [any, any, any]) => T) => T)
+    & (<T>((...args: Array<any>) => T) => T)
```

Comment:
I'm curious about the use case here, could you provide a couple examples? It seems like line 209 is a catch-all, does 206-208 provide value then?

Reply:
Added some more tests samples of usages, since action thunks that are defined as spread array vs spread tuple are non-compatible they need to be catered separately. Tuples are generally more strongly typed since you can type each arg, but if someone was lazy to use array version I didn't want to block the ability to have return types defined which is real value of this change